topics/secret-management/index.md

```diff
@@ -0,0 +1,293 @@
+[//]: # (TODO: Secret cycling and caching/refreshing values in App Configuration)
+[//]: # (TODO: Interaction with deploy templates)
+
+## Getting it to work {id=getting-it-to-work}
+
+### Basic Azure Key Vault integration
+
+Most secrets are usually treated as configuration values and to facilitate that, `IConfigurationBuilder` can be configured to read from an Azure Key Vault.
+
+``` csharp
+Host
+    .CreateDefaultBuilder(args)
+    .ConfigureAppConfiguration((context, config) =>
+    {
+        var azureKeyVaultEndpoint = GetAzureKeyVaultEndpoint(context.HostingEnvironment.EnvironmentName);
+        var keyVaultClient = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(new AzureServiceTokenProvider().KeyVaultTokenCallback));
+        config.AddAzureKeyVault(azureKeyVaultEndpoint, keyVaultClient, new DefaultKeyVaultSecretManager());
+    });
+```
+
+Requires `Microsoft.Extensions.Configuration.AzureKeyVault` {.code-annotation}
+
+The `AzureServiceTokenProvider` is a neat class that tries a couple of authentication strategies before it fails. Locally that includes the Azure Service Authentication account you configured in Visual Studio and deployed as a Web Service or Function App it will attempt to use a managed identity. When you use a user-assigned managed identity, [the token provider needs to know the Client Id of the identity][Azure Key Vault authentication: managed identities]. You can pass that information in as part of a connection string, either passed to the constructor `new AzureServiceTokenProvider("RunAs=App;AppId={ClientId of user-assigned identity}")` or specified in the `AzureServicesAuthConnectionString` environment variable. Other scenario's are listed [here][Azure Key Vault authentication: connection strings].
+
+Using Azure Key Vault like this covers most cases. You still have to account for different environments and authentication scenario's, but this is the gist of it.  
+
+### Adding Azure App Configuration to the mix
+
+Integrating Azure Key Vault into Azure App Configuration is easy and replaces the Key Vault integration mentioned above.
+
+``` csharp
+var cred = new DefaultAzureCredential();
+
+return config.AddAzureAppConfiguration(options =>
+{
+    options
+        .Connect(appConfigEndpoint, cred)
+        .ConfigureKeyVault(kv =>
+        {
+            kv.SetCredential(cred);
+        })
+        .Select(KeyFilter.Any, LabelFilter.Null)
+        .Select(KeyFilter.Any, environmentName);
+});
+```
+
+Requires `Microsoft.Extensions.Configuration.AzureAppConfiguration` and `Azure.Identity` {.code-annotation}
+
+`DefaultAzureCredential` works a bit different than `AzureServiceTokenProvider` however. The most important difference I encountered is that when using an Azure Service Authentication account in Visual Studio that is a guest account to the Azure directory your resources are in, you need to explicitly configure the tenant id as well as use at least version `1.2.0-preview.2` of `Azure.Identity`.  
```

Comment:
1. Is this only true in visual studio or also in when you have this service deployed?
2. Is this only true vs the Azure App Configuration resource, or any resource you want to authenticate against (like Key Vault)